src/main/haskell/kore/src/Kore/Exec.hs

```diff
@@ -0,0 +1,285 @@
+{-|
```

Comment:
Since you extracted this as a library, it would be nice to have a few tests.

Reply:
Yes, I was thinking I would hardcode a simple module and try some executions and searches.